pkg/util/yaml/decoder.go

```diff
@@ -52,10 +53,16 @@ type YAMLToJSONDecoder struct {
 // stream in chunks by converting each document (as defined by
 // the YAML spec) into its own chunk, converting it to JSON via
 // yaml.YAMLToJSON, and then passing it to json.Decoder.
-func NewYAMLToJSONDecoder(r io.Reader) *YAMLToJSONDecoder {
+func NewYAMLToJSONDecoder(r io.Reader, bytes []byte, isStdin bool) *YAMLToJSONDecoder {
 	reader := bufio.NewReader(r)
-	return &YAMLToJSONDecoder{
-		reader: NewYAMLReader(reader),
+	if isStdin && sessionIsTTY() {
+		return &YAMLToJSONDecoder{
+			reader: NewYAMLByteReader(bytes),
```

Comment:
Couldn't find another way to distinguish if it's a JSON stream in a TTY context. The `Read*` methods all have the same issue where they are waiting for a `io.EOF` character to unblock.

Not exactly sure where to go in this PR from here ?